### ISSUE TITLE
Does Ruff fail to format parts of a file when syntax errors are present?

### ISSUE THREAD

**Issue by work4lazy** [astral-sh/ruff#15166]
Hello,

I am using the Ruff extension in VSCode to automatically format Python files. However, I have noticed that when I encounter a severe syntax error while writing code, Ruff is unable to properly format any part of this file.

I would like to ask if this behavior is expected: that a severe syntax error in one place prevents Ruff from formatting all other parts of the file correctly?

Thank you for your assistance!

**Comment by MichaReiser** [comment]
Yes, this is the expected behavior. We may explore formatting files with syntax errors in the future but it isn't on our immediate roadmap. 

Formatting files with syntax errors comes with its own challenges because the formatter should avoid changing your code to something you didn't intend. But we could try to at least format blocks that contain no syntax errors.
